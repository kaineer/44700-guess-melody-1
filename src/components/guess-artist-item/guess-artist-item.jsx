import React from 'react';
import {string, number, func} from 'prop-types';

export const GuessArtistItem = ({artist, picture, orderId, onClick}) => (
  <div className="artist">
    <input className="artist__input visually-hidden" type="radio" name="answer" value={`artist-${orderId}`} id={`answer-${orderId}`}
      onClick={onClick}
    />
    <label className="artist__name" htmlFor={`answer-${orderId}`}>
      <img className="artist__picture" src={picture} alt={artist} />
      {artist}
    </label>
  </div>
);

GuessArtistItem.propTypes = {
  artist: string.isRequired,
  picture: string.isRequired,
  orderId: number.isRequired,
  onClick: func
};
