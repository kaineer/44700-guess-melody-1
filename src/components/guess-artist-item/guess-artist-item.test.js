import React from 'react';
import {GuessArtistItem} from './guess-artist-item';
import {create} from 'react-test-renderer';

const question = {
  type: `artist`,
  song: {
    artist: `One`,
    src: ``
  },
  answers: [
    {
      picture: ``,
      artist: `One`
    },
    {
      picture: ``,
      artist: `Two`
    }
  ]
};

describe(`GuessArtistItem`, () => {
  it(`should render correctly`, () => {
    const {answers: [{picture, artist}]} = question;
    const clickHandler = () => null;

    const tree = create(
      <GuessArtistItem
        artist={artist}
        picture={picture}
        orderId={42}
        key={42}
        onClick={clickHandler}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
