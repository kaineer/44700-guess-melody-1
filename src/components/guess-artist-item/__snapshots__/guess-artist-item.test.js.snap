// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuessArtistItem should render correctly 1`] = `
<div
  className="artist"
>
  <input
    className="artist__input visually-hidden"
    id="answer-42"
    name="answer"
    onClick={[Function]}
    type="radio"
    value="artist-42"
  />
  <label
    className="artist__name"
    htmlFor="answer-42"
  >
    <img
      alt="One"
      className="artist__picture"
      src=""
    />
    One
  </label>
</div>
`;
